pages/home.jsx

```diff
@@ -146,11 +149,121 @@ var BlogSection = React.createClass({
+  var errors = [];
+  // regex copied from http://stackoverflow.com/a/46181
+  var regexEmail = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
+  if (!regexEmail.test(signUpFormState.email)) {
+    errors.push("Please enter an email address.");
+  }
+
+  return errors;
+};
+
+var ModalPledge = React.createClass({
+  mixins: [React.addons.LinkedStateMixin],
+  getInitialState: function() {
+    return {
+      email: "",
+      validationErrors: []
+    };
+  },
+  handleSubmit: function(e) {
+    var validationErrors = validateSignupForm(_.pick(this.state,"email"));
+
+    if (validationErrors.length) {
+      e.preventDefault();
+      this.setState({validationErrors: validationErrors});
+      return;
+    }
+
+    if (process.env.NODE_ENV !== 'production' &&
+        !process.env.PLEDGE_MAILINGLIST_URL) {
+      e.preventDefault();
+      alert("PLEDGE_MAILINGLIST_URL is not defined. Simulating " +
+            "a successful pledge signup now.");
+      window.location = "?pledge=thanks";
+    }
+  },
+  renderValidationErrors: function() {
+    if (this.state.validationErrors.length) {
+      return (
+        <div className="alert alert-danger">
+          <p className="error-msg">Please enter an email address.</p>
+        </div>
+      );
+    }
+  },
+  render: function() {
+    return (
+      <Modal modalTitle="" foldedStyle className="modal-pledge">
+        <ImageTag className="image center-block"
+                  src1x="/img/pages/home/svg/icon-teach-man-chalkboard-pledge.svg"
+                  alt="" width={150} height={150} />
+        <p>Because the Web is a global public resource that's integral to modern life, <strong>I pledge to:</strong></p>
+        <ul>
+          <li>teach others digital literacy skills through hands-on making</li>
+          <li>help people move beyond simply consuming the web to creating it</li>
+          <li>be a passionate advocate of an open and accessible web</li>
+        </ul>
+        <p><strong>Pledge now, and we'll follow up with with details about how you can teach the web!</strong></p>
+        <form className="mailinglist-signup" action={process.env.PLEDGE_MAILINGLIST_URL} method="POST" onSubmit={this.handleSubmit}>
+          <fieldset>
+            <label htmlFor="pledge-email" className="sr-only">email</label>
+            <div className="icon-field-container">
+              <i className="fa fa-envelope"></i>
+              <input id="pledge-email" name="email" type="email" size="30" placeholder="email@example.com" valueLink={this.linkState("email")} required />
+            </div>
+            <label htmlFor="pledge-pp-note" className="sr-only">I'm okay with you handling this info as you explain in your <a href="https://www.mozilla.org/en-US/privacy/websites/">privacy policy</a></label>
+            <input id="pledge-pp-note" name={process.env.PLEDGE_MAILINGLIST_PRIVACY_NAME} type="checkbox" checked readOnly required hidden />
```

Comment:
Hmm, because of the `hidden` attribute on this checkbox, I'm not sure if screen readers will actually be able to announce its label to users. I wonder if it might be better to give it a `sr-only` class instead of `hidden`?

Also, might want to put the label text in a `<span>` variable to avoid reduplication but it's up to you. Another alternative might be to remove the `<label>`, give the `.pp-note` below an `id`, and map that to the [`aria-labelledby` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute) of the checkbox, though it might not work on really old screen readers...